client/lib/recommended-sites-store/store.js

```diff
@@ -42,7 +42,7 @@ function receiveRecommendations( data ) {
 		recommendations = recommendations.union( fromJS( pruned ) );
 
 		if ( recommendations !== previousRecs ) {
-			if ( recommendations.length >= MAX_RECOMMENDATIONS ) {
+			if ( recommendations.count >= MAX_RECOMMENDATIONS ) {
```

Comment:
Sure, we're pretty light on tests for this store right now. Any help would be appreciated. :)